azavea/iow-boundary-tool
# Add municipal boundaries layer

## Overview

This PR adds the municipal boundaries layer. The data is sourced from https://linc.osbm.nc.gov/explore/dataset/municipalities-2020/.

~The GeoJSON file is compressed with weback and then dynamically imported using the technique described here: https://www.azavea.com/blog/2022/04/28/how-to-code-split-geojson-layers-for-your-mapboxgl-js-react-apps/.~

The GeoJSON file is placed in the public folder and is retrieved with `fetch`.

Closes #34 

### Notes

The GeoJSON file compresses to about 4mb. If this is too much or the rendering of the entire file causes choppiness, then an alternative approach might be needed. A possibility is to split the layer into tiles and serve it from AWS.

## Testing Instructions

- Go to the draw page
- The municipal boundaries shoud show up.
- Clicking the sidebar visibility button should toggle the visibility of the layer

## Checklist
